### PR TITLE
KTOR-3352 Rewrite CallId to New Plugins API

### PR DESCRIPTION
**Pull request by Ololoshechkin**
https://youtrack.jetbrains.com/issue/KTOR-3352

### CODE REVIEW

File: ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/application/CreatePluginUtils.kt [hunk 1]
```diff
@@ -193,5 +198,6 @@ private fun <Configuration : Any, Plugin : ApplicationPluginBuilder<Configuratio
         it.action(pipeline.sendPipeline)
     }
 
-    hooks.forEach { it.install(application) }
+    println("   installing hooks: ${hooks.size}")
```
Comment (rsinukov) [inline ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/application/CreatePluginUtils.kt:201]: Please remove `println`

**Review by rsinukov — APPROVED** [review]

**Comment by rsinukov** [comment]
Please update api dump

File: ktor-server/ktor-server-plugins/ktor-server-call-id/jvm/src/io/ktor/server/plugins/CallId.kt [hunk 2]
```diff
@@ -40,207 +38,194 @@ public class RejectedCallIdException(
 }
 
 /**
- * A call id that is retrieved or generated by [CallId] plugin or `null` (this is possible if there is no
- * call id provided and no generators configured or [CallId] plugin is not installed)
+ * [CallId] plugin's configuration
  */
-public val ApplicationCall.callId: String? get() = attributes.getOrNull(CallId.callIdKey)
+public class CallIdConfiguration {
```
Comment (olme04) [inline ktor-server/ktor-server-plugins/ktor-server-call-id/jvm/src/io/ktor/server/plugins/CallId.kt:43]: as from https://github.com/ktorio/ktor/pull/2782#discussion_r787789014:
> We agree to rename Configuration to Config everywhere and put prefixes with a clear name.

Comment (Ololoshechkin) [reply ktor-server/ktor-server-plugins/ktor-server-call-id/jvm/src/io/ktor/server/plugins/CallId.kt]: Thanks for pointing this out, changed to `CallIdConfig`

**Review by e5l — APPROVED** [review]

**Comment by e5l** [comment]
Please check tests

**Pull request merged**